src/gui/maptools/qgsmaptooleditblanksegments.cpp

```diff
@@ -0,0 +1,947 @@
+          {
+            mSymbolLayer = dynamic_cast<const QgsTemplatedLineSymbolLayerBase *>( sl );
+            mSymbolLayerIndex = idx;
+            return false;
+          }
+
+          // recurse over sub symbols
+          if ( const QgsSymbol *subSymbol = const_cast<QgsSymbolLayer *>( sl )->subSymbol();
+               subSymbol && !visitSymbol( subSymbol, leafIdentifier ) )
+          {
+            return false;
+          }
+        }
+
+        return true;
+      }
+
+      bool visit( const QgsStyleEntityVisitorInterface::StyleLeaf &leaf ) override
+      {
+        if ( leaf.entity && leaf.entity->type() == QgsStyle::SymbolEntity )
+        {
+          auto symbolEntity = static_cast<const QgsStyleSymbolEntity *>( leaf.entity );
+          if ( symbolEntity->symbol() )
+            visitSymbol( symbolEntity->symbol(), leaf.identifier );
+        }
+        return true;
+      }
+
+      const QgsSymbol *symbol() const { return mSymbol; }
+      const QgsTemplatedLineSymbolLayerBase *symbolLayer() const { return mSymbolLayer; }
+      int symbolLayerIndex() const { return mSymbolLayerIndex; }
+
+      bool found() const { return mSymbol && mSymbolLayer; }
+
+    private:
+      const QgsSymbol *mSymbol = nullptr;
+      const QgsTemplatedLineSymbolLayerBase *mSymbolLayer = nullptr;
+      int mSymbolLayerIndex = -1;
+      const QString &mSymbolLayerId;
+  };
+
+} //namespace
+///@endcond
+
+QgsMapToolEditBlankSegmentsBase::QgsMapToolEditBlankSegmentsBase( QgsMapCanvas *canvas, QgsVectorLayer *layer, QgsLineSymbolLayer *symbolLayer, int blankSegmentFieldIndex )
+  : QgsMapTool( canvas )
+  , mLayer( layer )
+  , mSymbolLayerId( symbolLayer->id() )
+  , mBlankSegmentsFieldIndex( blankSegmentFieldIndex )
+  , mEditedBlankSegment( new BlankSegment( canvas, mPoints ) )
+  , mStartRubberBand( new QgsRubberBand( canvas, Qgis::GeometryType::Point ) )
+  , mEndRubberBand( new QgsRubberBand( canvas, Qgis::GeometryType::Point ) )
+{
+  auto initRubberBand = []( QgsRubberBand *rb ) {
+    rb->setWidth( QgsGuiUtils::scaleIconSize( 4 ) );
+    rb->setColor( QgsSettingsRegistryCore::settingsDigitizingLineColor->value() );
+    rb->setIcon( QgsRubberBand::IconType::ICON_BOX );
+  };
+
+  initRubberBand( mStartRubberBand );
+  initRubberBand( mEndRubberBand );
+
+  mEditedBlankSegment->setHighlighted( true );
+}
+
+QgsMapToolEditBlankSegmentsBase::~QgsMapToolEditBlankSegmentsBase() = default;
+
+void QgsMapToolEditBlankSegmentsBase::activate()
+{
+  if ( !mLayer || !mLayer->renderer() )
+    return;
+
+  if ( !mSymbol || !mSymbolLayer )
+  {
+    // search and symbol and symbol layer
+    SymbolLayerVisitor visitor( mSymbolLayerId );
+    mLayer->renderer()->accept( &visitor );
+    if ( visitor.symbol() && visitor.symbolLayer() && visitor.symbolLayerIndex() > -1 )
+    {
+      mSymbol.reset( visitor.symbol()->clone() );
+      if ( mSymbolLayer = createRenderedPointsSymbolLayer( visitor.symbolLayer() ); mSymbolLayer )
```

Comment:
ok, this is official: I am not fond of this `if` usage! But this is only a aesthetic remark, I do not know if this is common or *accepted* elsewhere in QGIS code. Furthermore, I think you can just do: 

```suggestion
      if ( mSymbolLayer = createRenderedPointsSymbolLayer( visitor.symbolLayer() ) )
```